Done.

manualcheg opened the pull request:
Created generic interface "ListDiffInterface<T>" with methods: theSameAs(),

equals().
Method theSameAs() realized in each class wich described elements of list.

Created generic class "BaseListDiffCallback<T extends ListDiffInterface<T>> extends DiffUtil.ItemCallback<T>".

Changed ListDiffCallback to BaseListDiffCallback<> in initialization "differ" in each nessesary classes.

Пришлось скопировать файлы сделанного мной проекта в новый форк из первого форка из-за моего косяка с push в main-ветку.

manualcheg committed:
Create generic interface "ListDiffInterface<T>" with methods: theSameAs(),
equals().
Method theSameAs() realized in each class wich described elements of list.

Created generic class "BaseListDiffCallback<T extends ListDiffInterface<T>> extends DiffUtil.ItemCallback<T>".

Changed ListDiffCallback to BaseListDiffCallback<> in initialization "differ" in each nessesary classes.

faritowich commented on app/src/main/java/ru/yandex/practicum/contacts/presentation/base/BaseListDiffCallback.java:
```diff
@@ -0,0 +1,16 @@
+    public boolean areContentsTheSame(@NonNull T oldItem, @NonNull T newItem) {
+        return oldItem.equals(newItem);
+    }
+}
```
Не хватает метода getChangePayload.

manualcheg replied:
Да, действительно. Поправил.

faritowich commented on app/src/main/java/ru/yandex/practicum/contacts/presentation/base/ListDiffInterface.java:
```diff
@@ -0,0 +1,6 @@
+
+public interface ListDiffInterface<T> {
+    public boolean theSameAs(T t);
+    public boolean equals(Object o);
```
Можно убрать модификаторы public, т.к. эти методы и так неявно публичны (Студия подсвечивает излишние модификаторы серым цветом и дает подсказку)

manualcheg replied:
Понял и убрал.

faritowich commented on app/src/main/java/ru/yandex/practicum/contacts/presentation/filter/model/FilterContactTypeUi.java:
```diff
@@ -20,6 +22,12 @@ public boolean isSelected() {
+
+    @Override
+    public boolean theSameAs(FilterContactTypeUi f) {
+        return this.getContactType() == f.getContactType();
```
Лучше использовать такой нейминг, чтобы по названию было понятно, что параметр означает. Например, newItem вместо f

manualcheg replied:
Хорошо. Исправил.

manualcheg committed:
delete some modifiers unwanted in "ListDiffInterface"

add lost method getChangePayload in "BaseListDiffCallback"

change name of variable in "FilterContactTypeUi"

manualcheg commented:
to create new pull request
